src/zjs_timers.c

```diff
@@ -108,6 +108,22 @@ static bool delete_timer(int32_t id)
     return false;
 }
 
+void zjs_timers_free_timers()
+{
+    for (zjs_timer_t **ptm = &zjs_timers; *ptm; ptm = &(*ptm)->next) {
+        zjs_timer_t *tm = *ptm;
+        int i;
+        zjs_port_timer_stop(&tm->timer);
+        *ptm = tm->next;
```

Comment:
I'm pretty sure this is buggy. E.g. the first time you're looking at zjs_timers (the first list object) and here you set *ptm (aka zjs_timers) to the second list object. After the loop iteration, the "next" statement runs which sets ptm to the pointer to the third list object, hence skipping the second.

The delete function worked because it returned after doing this line, it didn't continue in the loop.

You can simplify this whole thing by implementing the function as:
while (zjs_timers)
    delete_timer(zjs_timers->callback_id)

Slightly inefficient, but pretty obviously correct.

Or at least replace the outer for statement with a while (zjs_timers) to make sure you're not skipping any.